relay-monitors/src/lib.rs

```diff
@@ -76,6 +118,10 @@ struct CheckIn {
     /// Duration of this check since it has started in seconds.
     #[serde(default, skip_serializing_if = "Option::is_none")]
     duration: Option<f64>,
+
+    /// monitor configuration to support upserts.
+    #[serde(default, skip_serializing_if = "Option::is_none")]
+    monitor_config: Option<MonitorConfig>,
```

Comment:
This is a little tricky, since the monitor object actually has a `config` as it's own property. 

https://github.com/getsentry/sentry/blob/ad9fa09acb39820a12908d57e0b04a29c71b3a45/src/sentry/monitors/models.py#L175

There are other fields on a monitor such as it's name and project that are NOT part of the config. I wanted to keep it this way for consistency.

That said, from an API ingest pov we could simply call this `monitor` and it would be OK, let me bring it up and see what everyone think.s